project/Dependencies.scala

```diff
@@ -113,4 +113,11 @@ object Dependencies {
 
   val scalaXml = scala211Module("scala-xml", "1.0.6")
   val scalaParsers = scala211Module("scala-parser-combinators", "1.0.5")
+
+  def log4jVersion = "2.8.1"
+  val log4jApi = "org.apache.logging.log4j" % "log4j-api" % log4jVersion
+  val log4jCore = "org.apache.logging.log4j" % "log4j-core" % log4jVersion
+  val log4jSlf4jImpl = "org.apache.logging.log4j" % "log4j-slf4j-impl" % log4jVersion
+  // specify all of log4j modules to prevent misalignment
+  val log4jDependencies = Vector(log4jApi, log4jCore, log4jSlf4jImpl)
```

Comment:
(The source of my question was I thought we were already doing this in sbt/librarymanagement, but instead we declare but don't use log4j-slf4j-impl there: https://github.com/sbt/util/blob/v1.0.0-M28/project/Dependencies.scala#L54)

I still don't fully understand the failure in #3333, but I think trying to fix it like this is risky. We should check what happens with this in sbt-web/sbt-play where they prefer logback as the slf4j impl.